.github/workflows/clp-core-build.yaml

```diff
@@ -73,14 +73,14 @@ jobs:
           local_registry_port: "5000"
           token: "${{secrets.GITHUB_TOKEN}}"
 
-  ubuntu-bionic:
+  ubuntu-focal:
```

Comment:
Is there no way to simplify all the distributions? It seems like they're almost the exact same code repeated

Reply:
Yeah, I think we could collapse the checkout, workaround, and filter steps into the `clp-core-build-deps-and-binaries` step. The reason I didn't do it initially was because it felt like the checkout step should be a direct child of the workflow so that it's clear where the code gets checked out. Then the work around is directly related to the checkout step, so I kept that one. And then the filter was just a judgement call, but I think that could be moved into the action.

We could try it in another PR. Probably too big of a change for this PR.